### ISSUE TITLE
Double Placeholder Potential Problem

### ISSUE THREAD

**Issue by mkhatib** [ampproject/amphtml#2830]
@dvoytenko pointed out that in `amp-youtube` or other elements that generates a dynamic placeholder if they detect no placeholder already is provided, there's a potential problem that the placeholder haven't yet been parsed and hence amp-youtube elements will have 2 placeholder elements.

This is slightly related to #1014 with the exception that placeholders are managed by the runtime and not the element itself.

We chatted about different ways to fix this, one simple fix for this is just to `display:none` the dynamicly-created extra placeholder once the new placeholder shows up (the browser can handles this not us).

On the same issue we chatted about moving managing this to resources after the build phase instead of leaving this to the elements themselves. Something like the following snippet (shamelessly stolen from @dvoytenko)

```Javascript
build(force) {
   ...
   this.element.build(force);
   ...
    if (!this.element.getPlaceholder()) {
      var placeholder = this.element.createPlaceholder();
      this.element.appendChild(placeholder);
      // <x>
      //   <div auto-generated placeholder>
      //   <div user-defined placeholder>
      // </x>
    }
```

```CSS
.-amp-element > [placeholder]:first-child {
  display: none;
}
.-amp-element > [placeholder]:last-child {
  display: initial;
}
```

Or something like that

**Comment by dvoytenko** [comment]
@mkhatib Makes sense. Although this probably won't be `:last-child` since that literally selectors the last child in the container. Possible a sibling selector or maybe `:last-of-type` would do.

**Comment by rudygalfi** [comment]
@mkhatib Should we move this into the next sprint?

**Comment by mkhatib** [comment]
Yeah I think this will have to move for the next sprint. Just sent a change that addresses this but I it still might need some back and forth.

**Comment by ericlindley-g** [comment]
Rolling into the next milestone

**Comment by mkhatib** [comment]
This has been fixed with #3174
